app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -266,16 +271,19 @@ public function getCartDataJs($checkoutType = self::CHECKOUT_TYPE_MULTI_PAGE)
 
 
     /**
-     * Calculates and returns the key for storing a quote's Bolt cart
+     * Calculates and returns the key for storing a Bolt order token
      *
-     * @param Mage_Sales_Model_Quote $quote         The quote whose key that will be generated
-     * @param string                 $checkoutType  'multi-page' | 'one-page' | 'admin'
+     * @param Mage_Sales_Model_Quote|array $quote         The quote whose key that will be
+     *                                                    generated or the bolt cart array
+     *                                                    repesentation of the quote.
      *
-     * @return string   The calculated key for this quote's cart.  Format is {quote id}_{md5 hash of cart content}
+     * @param string                       $checkoutType  'multi-page' | 'one-page' | 'admin'
+     *
+     * @return string   The calculated key for this quote's cart.
+     *                  Format is {quote id}_{md5 hash of cart content}
      */
     protected function calculateCartCacheKey( $quote, $checkoutType ) {
-        $boltHelper = Mage::helper('boltpay/api');
-        $boltCartArray = $boltHelper->buildOrder($quote, $checkoutType === 'multi-page');
+        $boltCartArray = is_array($quote) ?: Mage::getModel('boltpay/boltOrder')->buildOrder($quote, $checkoutType === 'multi-page');
```

Comment:
If passed $quote variable is an array (the bolt cart repesentation) shoud the $boltCartArray be that array? This way it is a bool(true).
```
$boltCartArray = is_array($quote) ? $quote : Mage::getModel('boltpay/boltOrder')->buildOrder($quote, $checkoutType === 'multi-page');
```

Reply:
@danac-gs Yep.  Good catch!